src/Compilers/CSharp/Portable/Compiler/TypeCompilationState.cs

```diff
@@ -164,6 +172,58 @@ public void Free()
             }
 
             _wrappers = null;
+            _constructorInitializers = null;
+        }
+
+        /// <summary>
+        /// Report an error if adding the edge (method1, method2) to the ctor-initializer
+        /// graph would add a new cycle to that graph.
```

Comment:
A new cycle is one that wasn't previously there. I use this word for clarity, since the call graph may have had cycles among previously considered edges, even though such cycles are not recorded in the graph that we store as we avoid adding cycle-inducing edges.